docs/samples/bentoml/README.md

```diff
@@ -0,0 +1,189 @@
+    saved_path = iris_classifier_service.save()
+```
+
+The sample code above can be found in the BentoML repository, run them directly with the
+following command:
+
+```bash
+git clone git@github.com:bentoml/BentoML.git
+python ./bentoml/guides/quick-start/main.py
+```
+
+After saving the BentoService instance, you can now start a REST API server with the
+model trained and test the API server locally:
+
+```bash
+# Start BentoML API server:
+bentoml serve IrisClassifier:latest
+```
+
+```bash
+# Send test request:
+curl -i \
+  --header "Content-Type: application/json" \
+  --request POST \
+  --data '[[5.1, 3.5, 1.4, 0.2]]' \
+  http://localhost:5000/predict
+```
+
+### Deploy InferenceService
+
+BentoML provides a convenient way of containerizing the model API server with Docker. To
+create a docker container image for the sample model above:
+
+1. Find the file directory of the SavedBundle with `bentoml get` command, which is
+directory structured as a docker build context.
+2. Running docker build with this directory produces a docker image containing the API
+model server.
+
+```shell
+model_path=$(bentoml get IrisClassifier:latest -q | jq -r ".uri.uri")
+
+# Replace {docker_username} with your Docker Hub username
+docker build -t {docker_username}/iris-classifier $model_path
+docker push {docker_username}/iris-classifier
+```
+
+*Note: BentoML's REST interface is different than the Tensorflow V1 HTTP API that
+KFServing expects. Requests will send directly to the prediction service and bypass the
+top-level InferenceService.*
+
+*Support for KFServing V2 prediction protocol with BentoML is coming soon.*
+
+The following is an example YAML file for specifying the resources required to run an
+InferenceService in KFServing. Replace `{docker_username}` with your Docker Hub username
+and save it to `bentoml.yaml` file:
+
+```yaml
+apiVersion: serving.kubeflow.org/v1alpha2
+kind: InferenceService
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: iris-classifier
+spec:
+  default:
+    predictor:
+      custom:
+        container:
+          image: {docker_username}/iris-classifier
+          ports:
+            - containerPort: 5000
+```
+
+Use `kubectl apply` command to deploy the InferenceService:
+
+```shell
+kubectl apply -f bentoml.yaml
+```
+
+### Run prediction
+
+*Note: Use kfserving-ingressgateway as your INGRESS_GATEWAY if you are deploying
+KFServing as part of Kubeflow install, and not independently.*
+
+```shell
+MODEL_NAME=iris-classifier
+INGRESS_GATEWAY=istio-ingressgateway
+CLUSTER_IP=$(kubectl -n istio-system get service $INGRESS_GATEWAY -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+SERVICE_HOSTNAME=$(kubectl get inferenceservice ${MODEL_NAME} -o jsonpath='{.status.url}' | cut -d "/" -f 3)
+
+curl -v -H "Host: ${SERVICE_HOSTNAME}" \
+  --header "Content-Type: application/json" \
+  --request POST \
+  --data '[[5.1, 3.5, 1.4, 0.2]]' \
+  http://$CLUSTER_IP/model/predict
+```
```

Comment:
I think this will not work currently since our top level virtual service is expecting `/v1/models/$MODEL_NAME:predict`

Reply:
Ok, I will run the guide and make the changes.

Reply:
@yuzisun BentoML deployment is very similar to the prebuild-image example, I updated it to match with that example.  Using `kubectl get route` instead of `kubectl get inferenceservice` for getting `SERVICE_HOSTNAME`